master/internal/api_model.go

```diff
@@ -243,8 +252,13 @@ func (a *apiServer) PatchModel(
 
 func (a *apiServer) ArchiveModel(
 	ctx context.Context, req *apiv1.ArchiveModelRequest) (*apiv1.ArchiveModelResponse, error) {
+	getResp, err := a.GetModel(ctx, &apiv1.GetModelRequest{ModelName: req.ModelName})
```

Comment:
This looks strange.  It looks like you are invoking an API endpoint handler as if it were a helper function?

I think you should refactor the part that is a helper function into a standalone helper function and call it from `a.GetModel` and also the places where you call `a.GetModel`.

Reply:
we're missing the 'service' layer from the controller/service/repo pattern everywhere in determined (except users) so this is fairly common. that pattern exists for exactly this, otherwise it gets annoying if you e.g. ever need to change get model, or you do a refactor of a bunch of model related stuff or `GetModel` panics because it thinks its always a top level api handler, etc etc. it is some overhead (for devs), and so it's probably most useful to add this layer only when you start to do these sorts of calls, because this is the point we're you begin walking into a hole for not having it.

Reply:
@stoksc are you agreeing with my refactor suggestion or disagreeing and saying that it would be hurtful to do the refactor suggestion now?

Reply:
In this latest commit I separate out `ModelFromIdentifier` and `ModelVersionFromId` to be reusable helper functions instead of using the API calls:  https://github.com/determined-ai/determined/pull/3723/commits/219e8011df3a864a178b9997d4f14616d690ee45

From my perspective it's a small-footprint change, just let me know if there's a preference about whether these (and `clearModelName`) should be in another file or named differently.